### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by alexAubin** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by alexAubin** [comment]
At this stage i don't know what else to do but actually installing Roundcube locally and investigating what the hell happens when calling bin/update.sh (im even trying to read https://github.com/roundcube/roundcubemail/blob/master/bin/update.sh but can't understand how nor why composer is called exactly)

**Comment by CodeShakingSheep** [comment]
Just read about the update problem in this PR and it might be related to the `kolab/net_ldap3` plugin. I had a similar problem 3 weeks ago when I tried to run `composer.phar update` manually and it happened because the git repository https://git.kolab.org/ where the mentioned plugin is hosted was down. If I remember correctly it was down for several days and until it was back up again I couldn't run the update command because it always timed out. See my comment on the issue here: https://github.com/roundcube/roundcubemail/issues/7416

So, it might be worth a shot to run the tests again because now the git repository is up again.

**Comment by alexAubin** [comment]
Hmm I'm not sure this is the same issue, from what we can see on the CI, the script launches the bin/update.sh script, and then it hangs forever for seemingly no reason ...

Or maybe we need to upgrade to a newer version that contains the fix you mention if this is really the root cause of the issue

**Comment by nicofrand** [comment]
The issue comes from composer:

```
Info: [#################+..] > Updating roundcube configuration...
Warning: Do not run Composer as root/super user! See https://getcomposer.org/root for details
Warning: In Factory.php line 693:
Warning:   The HOME or COMPOSER_HOME environment variable must be set for composer to
Warning:   run correctly
Warning: update [--with WITH] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--dry-run] [--dev] [--no-dev] [--lock] [--no-install] [--no-autoloader] [--no-suggest] [--no-progress] [-w|--with-dependencies] [-W|--with-all-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>...]
Info: [##################+.] > Updating roundcube core...
Warning: Do not run Composer as root/super user! See https://getcomposer.org/root for details
```

When launched manually it is indeed interactive (even though `-y` is set).

```
root@nicofrand:/var/www/roundcube# grep RCMAIL_VERSION program/include/iniset.php
define('RCMAIL_VERSION', '1.6.5');
root@nicofrand:/var/www/roundcube# ./bin/update.sh --version "1.6.0" -y
Executing database schema update.
Executing php composer.phar to update dependencies...
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]? yes
```

**Comment by nicofrand** [comment]
Workaround to set before calling `bin/upgrade.sh`:

```
export COMPOSER_ALLOW_SUPERUSER=1
```

Tested on my instance and the upgrade completed.

File: scripts/upgrade [hunk 1]
```diff
@@ -261,8 +261,8 @@ then
     ynh_replace_string --match_string="^\s*// installed plugins" --replace_string="&\n $installed_plugins" --target_file="$final_path/config/config.inc.php"
 
     # Update JavaScript dependencies
-    (cd "$final_path"
-    /usr/bin/php$phpversion -q ./bin/update.sh -v ?)
+    pushd "$final_path"
+    ./bin/update.sh --version="?" -y <<< ""
```
Comment (nicofrand) [inline scripts/upgrade:265]: Why is it done here and then on line 279?

File: scripts/upgrade [hunk 2]
```diff
@@ -275,8 +275,8 @@ then
     #=================================================
     ynh_script_progression --message="Updating $app core..." --weight=4
 
-    ( cd "$final_path"
-    ynh_exec_warn ./bin/update.sh --version=$oldversion -y)
+    ynh_exec_warn ./bin/update.sh --version=$oldversion -y
```
Comment (nicofrand) [inline scripts/upgrade:278]:
```suggestion
    COMPOSER_ALLOW_SUPERUSER=1 ynh_exec_warn ./bin/update.sh --version=$oldversion -y
```

**Review by nicofrand — COMMENTED** [review]
`COMPOSER_ALLOW_SUPERUSER=1 yunohost app upgrade roundcube -u https://github.com/YunoHost-Apps/roundcube_ynh/tree/testing -F` is enough to run the update with this branch.

I don't know about `ynh_exec_warn` but that should be ok too?

File: scripts/upgrade (hunk 1 above)
Comment (nicofrand) [reply scripts/upgrade]: Still wonder why it's called twice...

Comment (Salamandar) [reply scripts/upgrade]: Not sure about the double call. But I added your fix and it seems to do the trick, thanks.

**Comment by Salamandar** [comment]
Fixed o/

I'm testing the upgrade on my server then merging this.

**Review by Salamandar — APPROVED** [review]

**Pull request merged**